### PR TITLE
[fea] create docs using vitepress

### LINKED ISSUE #179: [fea] setup `vitepress` docs

### PR DESCRIPTION
**Pull request by marco-souza**
# Descrição

This PR adds documentation to our project using vitepress

- https://vitepress.dev/

## Changes

> Detalhamento, se necessario

- **fea: setup docs using vitepress**
- **enh: biome ignore generated files**

## Notes

Usage:

```sh
bun run docs:dev     # dev docs
bun run docs:build   # build docs
bun run docs:preview # preview built docs
```

## Preview

image

This closes #179

### CODE REVIEW

File: biome.json [hunk 1]
```diff
@@ -5,6 +5,14 @@
...
   "formatter": {
     "include": ["**/*.ts", "**/*.tsx"],
+    "ignore": [
+      "docs/.vitepress",
+      "node_modules",
+      ".next",
+      "packages/components/icons",
+      "**/*.(svg,js)",
+      "**/*/firestore.ts"
+    ],
```
Comment (marco-souza) [inline biome.json:15]: I don't know if we have a way to share these ignore statements across `formatter` and `linter` configs

File: docs/.vitepress/config.mts
Comment (marco-souza) [inline docs/.vitepress/config.mts:1]: This is the main config file for `vitepress` and it was autogenerated with `bunx vitepress init`.

File: docs/api-examples.md
Comment (marco-souza) [inline docs/api-examples.md:1]: autogenerated

File: package.json [hunk 2]
```diff
@@ -1,61 +1,65 @@
 {
-	"name": "webapp",
```
Comment (marco-souza) [inline package.json:2]: my bad, vitepress installation script changes replaces `<tab>` by `<space>`

File: docs/index.md
Comment (marco-souza) [inline docs/index.md:1]: autogenerated - file that configures what will appear on the main page

File: docs/markdown-examples.md
Comment (marco-souza) [inline docs/markdown-examples.md:1]: autogenerated

**Review by Joel-leal — APPROVED** [review]

File: docs/markdown-examples.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+```js{4}
+export default {
+  data () {
+    return {
+      msg: 'Highlighted!'
+    }
+  }
+}
+```
```
Comment (vzsoares) [inline docs/markdown-examples.md:33]: That slide tool that i used also has this syntax :+1:

Comment (marco-souza) [reply docs/markdown-examples.md:33]: No luck. I only found this: https://github.com/slidevjs/docs

But good to know, we can probably implement it with a vite plugin, if needed

**Pull request merged**